pkg/verify/verify.go

```diff
@@ -283,9 +223,54 @@ func (v *releaseVerifier) Verify(ctx context.Context, releaseDigest string) erro
 		}
 		return fmt.Errorf("unable to locate a valid signature for one or more sources")
 	}
+
+	v.cacheVerification(releaseDigest, signedWith)
+
 	return nil
 }
 
+// Signatures returns a copy of any cached signatures that have been validated
+// so far. It may return no signatures.
+func (v *ReleaseVerifier) Signatures() map[string][][]byte {
+	copied := make(map[string][][]byte)
+	v.lock.Lock()
+	defer v.lock.Unlock()
+	for k, v := range v.signatureCache {
+		copied[k] = v
+	}
+	return copied
+}
+
+// hasVerified returns true if the digest has already been verified.
+func (v *ReleaseVerifier) hasVerified(releaseDigest string) bool {
+	v.lock.Lock()
+	defer v.lock.Unlock()
+	_, ok := v.signatureCache[releaseDigest]
+	return ok
+}
+
+const maxSignatureCacheSize = 64
+
+// cacheVerification caches the result of signature check for a digest for later retrieval.
+func (v *ReleaseVerifier) cacheVerification(releaseDigest string, signedWith [][]byte) {
+	v.lock.Lock()
+	defer v.lock.Unlock()
+
+	if len(signedWith) == 0 || len(releaseDigest) == 0 || v.signatureCache == nil {
```

Comment:
`NewReleaseVerifier` initializes the map and `hasVerified` lacks a `nil` guard.  Feels strange to have a `nil` guard here.

Reply:
hasVerified doesn't need a nil guard.  Read from map doesn't.